core/trino-main/src/main/java/io/trino/operator/join/ArrayPositionLinks.java

```diff
@@ -64,7 +63,11 @@ public PositionLinks create(List<JoinFilterFunction> searchFunctions)
                 @Override
                 public long checksum()
                 {
-                    return XxHash64.hash(Slices.wrappedIntArray(positionLinks));
+                    long hash = 0;
+                    for (int positionLink : positionLinks) {
+                        hash = XxHash64.hash(hash, positionLink);
```

Comment:
It probably doesn't matter, but be aware that this is not equivalent to the old code in terms of semantics and performance.

Reply:
We could add an `update()` method that accepts a `long`

Reply:
Yep, I agree to both comments.  The code should be fine as this is only used during spilling, and only used as a checksum.